webdriver/tests/perform_actions/wheel.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+
+from webdriver.error import NoSuchWindowException
+
+from tests.perform_actions.support.refine import filter_dict, get_events
+from tests.support.asserts import assert_move_to_coordinates
+
+
+def test_null_response_value(session, wheel_chain):
+    value = wheel_chain.scroll(0, 0, 0, 10).perform()
+    assert value is None
+
+
+def test_no_browsing_context(session, closed_window, wheel_chain):
+    with pytest.raises(NoSuchWindowException):
+        wheel_chain.scroll(0, 0, 0, 10).perform()
+
+
+def test_wheel_scroll(session, test_actions_scroll_page, wheel_chain):
+    session.execute_script("document.scrollingElement.scrollTop = 0")
+
+    outer = session.find.css("#outer", all=False)
+    wheel_chain.scroll(0, 0, 5, 10, origin=outer).perform()
+    events = get_events(session)
+    assert len(events) > 0
+    event_types = [e["type"] for e in events]
+    assert "wheel" in event_types
+    for e in events:
+        if e["type"] == "wheel":
+            assert e["deltaX"] >= 5
+            assert e["deltaY"] >= 10
+            assert e["deltaZ"] == 0
+            assert e["target"] == "outer"
+
+
+def test_wheel_scroll_overflow(session, test_actions_scroll_page, wheel_chain):
+    session.execute_script("document.scrollingElement.scrollTop = 0")
+
+    scrollable = session.find.css("#scrollable", all=False)
+    wheel_chain.scroll(0, 0, 5, 10, origin=scrollable).perform()
+    events = get_events(session)
+    assert len(events) > 0
+    event_types = [e["type"] for e in events]
+    assert "wheel" in event_types
+    for e in events:
+        if e["type"] == "wheel":
+            assert e["deltaX"] >= 5
+            assert e["deltaY"] >= 10
+            assert e["deltaZ"] == 0
+            assert e["target"] == "scrollContent"
+
+
+def test_wheel_scroll_iframe(session, test_actions_scroll_page, wheel_chain):
+    session.execute_script("document.scrollingElement.scrollTop = 0")
+
+    subframe = session.find.css("#subframe", all=False)
+    wheel_chain.scroll(0, 0, 5, 10, origin=subframe).perform()
```

Comment:
So that means we operate on a sub-frame while not having switched to it before? From what I can see that shouldn't work, or is wheel scrolling different here? @jgraham?

Reply:
I have submitted a CL, which switched the frame https://github.com/web-platform-tests/wpt/blob/e949f1f611a601e82dca420c26d092fdd3475155/tools/webdriver/webdriver/client.py#L326, if the element is an iframe.

Reply:
So this was added by https://github.com/web-platform-tests/wpt/pull/17260 for finding an element via a CSS selector. Is that really something we are allowed to do? In my opinion that shouldn't have been merged. Nothing in `Find Element` should switch the frame. @jgraham please give your feedback here, which might be fixed in a follow-up PR.

Reply:
Oh wait. It's an additional option for the Python client. So it should be fine, but still hides frame switches in the tests, which why I was confused. Having to explicitly switch to a frame in the test should be better.